File: examples/python/heat2d/sample_posterior.py
#!/usr/bin/env python
import sys
import os.path
if not os.path.isfile('../../cxx/heat2d/model/libheat2d.so'):
 os.system('cd ../../cxx/heat2d/model && make')
sys.path.append('../../cxx/heat2d/model')
sys.path.append('../../../source/')
import libheat2d
import libkorali
 
korali = libkorali.Engine(libheat2d.heat2DSolver)

korali["Seed"] = 0xC0FFEE
korali["Verbosity"] = "Normal"

korali["Parameters"][0]["Name"] = "Intensity"
korali["Parameters"][0]["Type"] = "Computational"
korali["Parameters"][0]["Distribution"] = "Uniform"
korali["Parameters"][0]["Minimum"] = 10.0
korali["Parameters"][0]["Maximum"] = 60.0

korali["Parameters"][1]["Name"] = "PosX"
korali["Parameters"][1]["Type"] = "Computational"
korali["Parameters"][1]["Distribution"] = "Uniform"
korali["Parameters"][1]["Minimum"] = 0.0
korali["Parameters"][1]["Maximum"] = 0.5

korali["Parameters"][2]["Name"] = "PosY"
korali["Parameters"][2]["Type"] = "Computational"
korali["Parameters"][2]["Distribution"] = "Uniform"
korali["Parameters"][2]["Minimum"] = 0.6
korali["Parameters"][2]["Maximum"] = 1.0

korali["Parameters"][3]["Name"] = "Sigma"
korali["Parameters"][3]["Type"] = "Statistical"
korali["Parameters"][3]["Distribution"] = "Uniform"
korali["Parameters"][3]["Minimum"] = 0.0
korali["Parameters"][3]["Maximum"] = 20.0

korali["Problem"]["Objective"] = "Posterior"

p = libheat2d.heat2DInit()
for i in range(len(p.refTemp)):
 korali["Problem"]["Reference Data"][i] = p.refTemp[i]

korali["Solver"]["Method"] = "TMCMC"
korali["Solver"]["Covariance Scaling"] = 0.02
korali["Solver"]["Population Size"] = 3000

korali.run()

File: examples/heat2d/sample_posterior.cpp
#include "model/heat2d.hpp"
#include "korali.h"

int main(int argc, char* argv[])
{
 auto korali = Korali::Engine();

 korali["Seed"] = 0xC0FFEE;
 korali["Verbosity"] = "Normal";

 korali["Problem"]["Type"] = "Bayesian";
 korali["Problem"]["Variables"][0]["Name"] = "Intensity";
 korali["Problem"]["Variables"][0]["Type"] = "Computational";
 korali["Problem"]["Variables"][0]["Distribution"] = "Uniform";
 korali["Problem"]["Variables"][0]["Minimum"] = 10.0;
 korali["Problem"]["Variables"][0]["Maximum"] = 60.0;

 korali["Problem"]["Variables"][1]["Name"] = "PosX";
 korali["Problem"]["Variables"][1]["Type"] = "Computational";
 korali["Problem"]["Variables"][1]["Distribution"] = "Uniform";
 korali["Problem"]["Variables"][1]["Minimum"] = 0.0;
 korali["Problem"]["Variables"][1]["Maximum"] = 0.5;

 korali["Problem"]["Variables"][2]["Name"] = "PosY";
 korali["Problem"]["Variables"][2]["Type"] = "Computational";
 korali["Problem"]["Variables"][2]["Distribution"] = "Uniform";
 korali["Problem"]["Variables"][2]["Minimum"] = 0.6;
 korali["Problem"]["Variables"][2]["Maximum"] = 1.0;

 korali["Problem"]["Variables"][3]["Name"] = "Sigma";
 korali["Problem"]["Variables"][3]["Type"] = "Statistical";
 korali["Problem"]["Variables"][3]["Distribution"] = "Uniform";
 korali["Problem"]["Variables"][3]["Minimum"] = 0.0;
 korali["Problem"]["Variables"][3]["Maximum"] = 20.0;

 auto p = heat2DInit();
 for (size_t i = 0; i < p.refTemp.size(); i++)
	 korali["Problem"]["Reference Data"][i] = p.refTemp[i];

 korali["Solver"]["Method"] = "TMCMC";
 korali["Solver"]["Covariance Scaling"] = 0.02;
 korali["Solver"]["Population Size"] = 3000;

 korali.setModel([](Korali::ModelData& d) { heat2DSolver(d.getVariables(), d.getResults()); });

 korali.run();

 return 0;
}
